src/api/name.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Account } from "../core";
+import { RouterAddressLookup, mintDomainName } from "../internal/name";
+import { SingleSignerTransaction, GenerateTransactionOptions } from "../transactions/types";
+import { MoveAddressType } from "../types";
+import { AptosConfig } from "./aptos_config";
+
+/**
+ * A class to handle all `Coin` operations
+ */
+export class Name {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+
+    const routerAddress = config.ansRouter || RouterAddressLookup[config.network];
+    if (!routerAddress) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        [
+          "No ANS Router address found. This most likely means you are using",
+          "a custom network, a local network, or the devnet. If you plan on using",
+          "the Aptos Names API, you must provide a router address in the AptosConfig",
+          "(such as { network: Network.LOCAL, ansRouter: '0x123...' })",
+          "Note, there is no ANS contract published on devnet.",
+        ].join(" "),
+      );
+    }
```

Comment:
I don't want to block the use of custom networks, local, or devnet, that don't use the `Name` class. I decided to warn upon instantiation and throw errors if the user tries to call a class method without properly configuring the router.